--- lib/VMware/share/VMware/VICommon.pm	2025-04-24 17:18:24.938290503 +0200
+++ VICommon.pm	2025-04-24 17:18:18.690399614 +0200
@@ -2319,6 +2319,8 @@
    my $user_agent = $self->{user_agent};
    $user_agent->cookie_jar->as_string
       =~ m/(.*)vmware_soap_session=\"\\\"([0-9a-zA-Z-](.*)+)\\\"\"(.*)/;
+   $user_agent->cookie_jar->as_string
+      =~ m/(.*)vmware_soap_session=[\\\"]*([0-9a-zA-Z-]+)/ unless $2;
    return $2;
 }
