--- cpan.rb.orig	2026-02-04 17:44:02
+++ cpan.rb	2026-02-04 17:33:43
@@ -333,7 +333,7 @@
 
     # Search metacpan to get download URL for this version of the module
     metacpan_search_url = "https://fastapi.metacpan.org/v1/release/_search"
-    metacpan_search_query = '{"fields":["download_url"],"filter":{"term":{"name":"' + "#{distribution}-#{self.version}" + '"}}}'
+    metacpan_search_query = '{"query":{"bool":{"filter":[{"term":{"name":"' + "#{distribution}-#{self.version}" + '"}}]}},"fields": ["download_url"]}'
     begin
       search_response = httppost(metacpan_search_url,metacpan_search_query)
     rescue Net::HTTPServerException => e
